lib/plugins/aws/deployFunction.js

```diff
@@ -109,11 +158,23 @@ class AwsDeployFunction {
       params.KMSKeyArn = serviceObj.awsKmsKeyArn;
     }
 
-    if ('description' in functionObj && !_.isObject(functionObj.description)) {
+    if (params.KMSKeyArn && params.KMSKeyArn === remoteFunctionConfiguration.KMSKeyArn) {
+      delete params.KMSKeyArn;
+    }
+
+    if (
+      'description' in functionObj &&
+      !_.isObject(functionObj.description) &&
```

Comment:
I see it reflects the original logic, but as we're updating it, I would improve it:

1. In case of string property more natural to check for existence with `functionObj.description`
2. By schema this value can't be an object, so I would skip that check

Reply:
Great call, I believe I've applied it whenever it made sense and dropped checks for `isObject` in places where schema prevents it